### PR TITLE
Changes the Z-6 to be more of an Auto Rifle, and merges it with disposable launcher for CT.

### PR DESCRIPTION
**Pull request by Pariah919**
## About The Pull Request
Z-6 rebalanced, .25 fire rate. Pen multiplier added.
Merged with Disposable.

## Why It's Good For The Game
Makes the Disposable not leave CT without a longarm ~~which sort of ruins the entire disposable part lol~~ and makes the Z6 have a nice niche.

### CODE REVIEW

File: maps/torch/infantry/firearms.dm [hunk 1]
```diff
@@ -125,16 +125,17 @@
 
 //rifle
 /obj/item/weapon/gun/projectile/automatic/bullpup_rifle/sec/lmg
-	name = "Z6 Komodo"
+	name = "Z-6 Komodo Automatic Rifle"
```
Comment (BurpleBineapple) [inline maps/torch/infantry/firearms.dm]: I recommend keeping this as Z6 Komodo to fit with the other Z weapons. The LMG is called the Z2 Sheol, the rifle is called the Z9 Bulldog, so no need to add a dash and 'automatic rifle'

**Review by BurpleBineapple — CHANGES_REQUESTED** [review]

**Comment by ProbablyCarl** [comment]
Revert the name and I'll merge this, as per Pineapple's take.

File: maps/torch/infantry/firearms.dm (hunk 1 above)
Comment (Pariah919) [inline maps/torch/infantry/firearms.dm]:
```suggestion
	name = "Z6 Komodo"
```

**Review by Anton-Kr — COMMENTED** [review]
Unfortunately you got caught in a different change. `penetration_mult ` has been changed to `penetration_mod `  and is now additive. 

See https://github.com/BoHBranch/BoH-Bay/pull/1334 for details.

File: maps/torch/infantry/firearms.dm [hunk 2]
```diff
@@ -127,14 +127,15 @@
...
 	auto_eject = 1
 	one_hand_penalty = 8
-	jam_chance = 15 //frangible rounds might shatter if they're chambered improperly. Especially so with such a high firerate.
+	acc_mod_base = 3
+	penetration_mult = 1.1
 	firemodes = list(
-		list(mode_name="semi auto", burst=1, fire_delay=null, move_delay=null, use_launcher=null, one_hand_penalty=8, burst_accuracy=null, dispersion=null),
-		list(mode_name="burst fire",  burst=5, fire_delay=null, move_delay=2, one_hand_penalty=10, burst_accuracy=list(0,-1,-1), dispersion=list(0.0, 0.6, 1.0)),
+		list(mode_name="semi auto", burst=1, fire_delay=null, move_delay=null, use_launcher=null, one_hand_penalty=8, burst_accuracy=null, dispersion=list(0.5, 1)),
+		list(mode_name="automatic",  automatic = TRUE, fire_delay = 2,5, burst = 1, burst_delay = 0, move_delay=2, dispersion= 0),
```
Comment (Anton-Kr) [inline maps/torch/infantry/firearms.dm]: is that `2,5` meant to have a comma?

**Review by Anton-Kr — CHANGES_REQUESTED** [review]
see above

File: maps/torch/infantry/firearms.dm (hunk 2 above)
Comment (Anton-Kr) [reply maps/torch/infantry/firearms.dm]: also why the zero dispersion?

File: maps/torch/infantry/firearms.dm [hunk 3]
```diff
@@ -127,14 +127,15 @@
...
 	name = "Z6 Komodo"
 	desc = "The Hephaestus Industries Z6 Komodo is an old bullpup carbine conversion. \
-	It adds the possibility of automatic fire, though at the cost of likely jams."
+	It is equiped with a heavy barrel for combat, ensuring better accuracy and velocity for penetration at the cost of firerate and magazine size."
 	origin_tech = list(TECH_COMBAT = 4, TECH_MATERIAL = 3)
 	auto_eject = 1
 	one_hand_penalty = 8
-	jam_chance = 15 //frangible rounds might shatter if they're chambered improperly. Especially so with such a high firerate.
+	acc_mod_base = 3
+	penetration_mult = 1.1
```
Comment (BurpleBineapple) [inline maps/torch/infantry/firearms.dm]:
```suggestion
	penetration_mod = 3.5
```

Comment (BurpleBineapple) [reply maps/torch/infantry/firearms.dm]: This would fix the error. The armor pen of 5mmR is 35, so 10% extra would be 3.5.

**Review by BurpleBineapple — CHANGES_REQUESTED** [review]

File: maps/torch/infantry/firearms.dm [hunk 4]
```diff
@@ -127,14 +127,15 @@
...
 	name = "Z6 Komodo"
 	desc = "The Hephaestus Industries Z6 Komodo is an old bullpup carbine conversion. \
-	It adds the possibility of automatic fire, though at the cost of likely jams."
+	It is equiped with a heavy barrel for combat, ensuring better accuracy and velocity for penetration at the cost of firerate and magazine size."
 	origin_tech = list(TECH_COMBAT = 4, TECH_MATERIAL = 3)
 	auto_eject = 1
 	one_hand_penalty = 8
-	jam_chance = 15 //frangible rounds might shatter if they're chambered improperly. Especially so with such a high firerate.
+	acc_mod_base = 3
+	penetration_mod = 5
```
Comment (Pariah919) [inline maps/torch/infantry/firearms.dm]:
```suggestion
	penetration_mod = 3.5
```

**Review by Anton-Kr — APPROVED** [review]

**Pull request merged**